### PR TITLE
Add additional function so that x509 name with specific type can be added

### PR DESCRIPTION
**Pull request by JoshuaNitschke**
Fixes issue #1370

### CODE REVIEW

**Comment by JoshuaNitschke** [comment]
Looks like an [upgrade to cc crate](https://github.com/alexcrichton/cc-rs/commit/c668e9e4770fac31515fb999630c867e80efe2c0) is causing failures.

File: openssl/src/x509/mod.rs [hunk 1]
```diff
@@ -827,6 +827,33 @@ impl X509NameBuilder {
...
+    /// Add a field entry by str with a specific type. (ex: 19 for V_ASN1_PRINTABLESTRING)
+    ///
+    /// This corresponds to [`X509_NAME_add_entry_by_txt`].
+    ///
+    /// [`X509_NAME_add_entry_by_txt`]: https://www.openssl.org/docs/man1.1.0/crypto/X509_NAME_add_entry_by_txt.html
+    pub fn append_entry_by_text_with_type(
+        &mut self,
+        field: &str,
+        value: &str,
+        ty: i32,
```
Comment (sfackler) [inline openssl/src/x509/mod.rs:839]: This argument should be some kind of real type rather than a raw number.

Comment (JoshuaNitschke) [reply openssl/src/x509/mod.rs:839]: Just to make sure I understand: you would like an enum type defined?

Comment (sfackler) [reply openssl/src/x509/mod.rs:839]: Not exactly an enum type, but something like https://docs.rs/openssl/0.10.30/src/openssl/ssl/mod.rs.html#595

Comment (JoshuaNitschke) [reply openssl/src/x509/mod.rs:839]: Thanks for the example, I'll get a revision in either tonight or tomorrow.

**Comment by JoshuaNitschke** [comment]
@sfackler how's this?

**Comment by JoshuaNitschke** [comment]
@sfackler is there anything you'd like changed?  My org has a production app pointed to my branch, which is less than ideal, so I'm willing to make whatever changes needed to get this merged.   I appreciate your time.

File: openssl/src/x509/mod.rs [hunk 2]
```diff
@@ -878,6 +905,32 @@ impl X509NameBuilder {
         }
     }
 
+    /// Add a field entry by NID with a specific type.  See [Asn1Tag](openssl::asn1::Asn1Tag)
```
Comment (sfackler) [inline openssl/src/x509/mod.rs:908]: You should be able to have append_entry_by_nid above just call this method.

**Comment by sfackler** [comment]
Sorry for the delay!

**Comment by JoshuaNitschke** [comment]
Thanks @sfackler!  Happy Holidays!

**Pull request merged**